eZ/Publish/Core/Search/Common/Slot/AbstractSubtree.php

```diff
@@ -26,24 +29,28 @@ protected function indexSubtree($locationId)
         $subtreeIds = $locationHandler->loadSubtreeIds($locationId);
 
         foreach ($subtreeIds as $locationId => $contentId) {
-            $this->searchHandler->indexLocation(
-                $locationHandler->load($locationId)
-            );
-
-            if (isset($processedContentIdSet[$contentId])) {
-                continue;
+            if ($this->searchHandler instanceof LocationIndexer) {
+                $this->searchHandler->indexLocation(
+                    $locationHandler->load($locationId)
+                );
             }
 
-            $this->searchHandler->indexContent(
-                $contentHandler->load(
-                    $contentId,
-                    $contentHandler->loadContentInfo($contentId)->currentVersionNo
-                )
-            );
-
-            // Content could be found in multiple Locations of the subtree,
-            // but we need to (re)index it only once
-            $processedContentIdSet[$contentId] = true;
+            if ($this->searchHandler instanceof ContentIndexer || $this->searchHandler instanceof FullTextIndexer) {
+                if (isset($processedContentIdSet[$contentId])) {
+                    continue;
+                }
+
+                $this->searchHandler->indexContent(
+                    $contentHandler->load(
+                        $contentId,
+                        $contentHandler->loadContentInfo($contentId)->currentVersionNo
+                    )
+                );
```

Comment:
@alongosz I think you were right about slots that extend `AbstractSubtree` not being needed for Legacy Search Engine. Sorry for not checking that more carefully.

@andrerom 
> Shouldn't we then somehow let slots using AbstractSubtree specify this.

Refactored in https://github.com/ezsystems/ezpublish-kernel/pull/1716/commits/34d2d26e7e70a0140cfdca9e01fa24155b314c56, but with helper methods in the base `Slot`.
Now we index for full text in these slots:
- `CopyContent`
- `CopyContent`
- `CopyContent`
- `CopyContent`
- `CopyContent`